types/part_set.go

```diff
@@ -264,3 +264,20 @@ func (ps *PartSet) StringShort() string {
 	defer ps.mtx.Unlock()
 	return fmt.Sprintf("(%v of %v)", ps.Count(), ps.Total())
 }
+
+func (ps *PartSet) MarshalJSON() ([]byte, error) {
+	if ps == nil {
+		return []byte("nil-PartSet"), nil
```

Comment:
shouldn't this be a valid json? like `null`? or maybe `[]byte("\"nil-PartSet\"")`

Reply:
yes it should. maybe we should just do `{}`